js/examples/openai/frontend/app/components/transform.ts

```diff
@@ -6,6 +6,30 @@ export const isValidMessageData = (rawData: JSONValue | undefined) => {
   return true;
 };
 
+const getSpanIdFromAnnotations = (annotations?: JSONValue[]) => {
+  if (!annotations) return;
```

Comment:
yeah there is a fair amount of context here but the easiest way on the server to write something to the response that isn't text is via an annotation - so the server writes the spanId as an annotation on the message (theses are vercel ai-sdk annotations) then we parse it out to get it on the message that is passed around to ui components (that's whats happening here)

The crazy logic you commented on below is because annotations are of type JSONValue[] and JSONValue is of type string | number | null | {[x:string]: JSONValue} or something like that -

 i tried putting the span id in the header but it was swallowed an if you just write it to the stream you have to parse it out so it doesn't get rendered as apart of the message so this felt the cleanest